segments/hostname.sh

```diff
@@ -3,24 +3,33 @@
 
 TMUX_POWERLINE_SEG_HOSTNAME_FORMAT="${TMUX_POWERLINE_SEG_HOSTNAME_FORMAT:-short}"
 
+# check if user has hostname command
+has_hostname() {
+  command -v $1 >/dev/null
+}
+
 generate_segmentrc() {
-	read -r -d '' rccontents <<EORC
+  read -r -d '' rccontents <<EORC
 # Use short or long format for the hostname. Can be {"short, long"}.
 export TMUX_POWERLINE_SEG_HOSTNAME_FORMAT="${TMUX_POWERLINE_SEG_HOSTNAME_FORMAT}"
 EORC
-	echo "$rccontents"
+  echo "$rccontents"
 }
 
 run_segment() {
-	local opts=""
-	if [ "$TMUX_POWERLINE_SEG_HOSTNAME_FORMAT" == "short" ]; then
-		if shell_is_osx || shell_is_bsd; then
-			opts="-s"
-		else
-			opts="--short"
-		fi
-	fi
+  local opts=""
+  if [ "$TMUX_POWERLINE_SEG_HOSTNAME_FORMAT" == "short" ]; then
+    if shell_is_osx || shell_is_bsd; then
+      opts="-s"
+    else
+      opts="--short"
+    fi
+  fi
 
-	hostname ${opts}
-	return 0
+  if has_hostname hostname; then
+    hostname ${opts}
+  else
+    hostnamectl hostname
```

Comment:
I don't think we can assume that if `hostname(1)` is not in `$PATH`, that `hostnamectl` is gong to be be. We must check in the same way if this command exist.